app/src/main/java/com/support/android/designlibdemo/NightModeUtils.java

```diff
@@ -0,0 +1,65 @@
+package com.support.android.designlibdemo;
+
+import android.app.Activity;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AppCompatDelegate;
+import android.view.Menu;
+import android.view.MenuItem;
+
+class NightModeUtils {
+    private Activity activity;
+
+    private NightModeUtils(@NonNull Activity activity) {
+        this.activity = activity;
+    }
+
+    static NightModeUtils from(@NonNull Activity activity) {
+        return new NightModeUtils(activity);
+    }
+
+    boolean checkNightMode(@NonNull Menu menu) {
+        switch (AppCompatDelegate.getDefaultNightMode()) {
+            case AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM:
+                menu.findItem(R.id.menu_night_mode_system).setChecked(true);
+                break;
+            case AppCompatDelegate.MODE_NIGHT_AUTO:
+                menu.findItem(R.id.menu_night_mode_auto).setChecked(true);
+                break;
+            case AppCompatDelegate.MODE_NIGHT_YES:
+                menu.findItem(R.id.menu_night_mode_night).setChecked(true);
+                break;
+            case AppCompatDelegate.MODE_NIGHT_NO:
+                menu.findItem(R.id.menu_night_mode_day).setChecked(true);
+                break;
+            default:
+                return false;
+        }
+        return true;
+    }
+
+    boolean onNightModeItemSelected(@NonNull MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.menu_night_mode_system:
+                setNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM);
+                break;
+            case R.id.menu_night_mode_day:
+                setNightMode(AppCompatDelegate.MODE_NIGHT_NO);
+                break;
+            case R.id.menu_night_mode_night:
+                setNightMode(AppCompatDelegate.MODE_NIGHT_YES);
+                break;
+            case R.id.menu_night_mode_auto:
+                setNightMode(AppCompatDelegate.MODE_NIGHT_AUTO);
+                break;
+            default:
+                return false;
+        }
+        return true;
+    }
+
+    private void setNightMode(@AppCompatDelegate.NightMode int nightMode) {
+        AppCompatDelegate.setDefaultNightMode(nightMode);
+
+        activity.recreate();
```

Comment:
What about the parent activity that created this activity?  It might not be destroyed.  @chrisbanes this is an issue we ran into and I assume this is why it wasn't implemented.  We used startActivityForResult here, and then set the result code, and depending on that code returned, we restart the activity.  Activities before the parent wouldn't be impacted by this though.